### ISSUE TITLE
Query regarding Self Supervised learning model

### ISSUE THREAD

**Issue by Kartik17** [TRI-ML/packnet-sfm#31]
Hi, thanks for uploading the code. I had a few queries:
1. As we are specifying the camera intrinsics so I believe the pre-trained model won't give good results on datasets other than which they are trained?
2. Will the model learn depth in a scenario where there are very few distinct objects in the scene and environment that look like a repeating texture, eg: desert, farms, etc? Are these the situations or scenario where the model can benefit from some sort of supervision(Semi-supervised)
3. For the Custom Image model, since in config YAML file, we are specifying the resolution of the image, do we need to account for the new resolution in the dummy camera matrix manually or there is function somewhere which premultiplies the camera matrix with the scale matrix.

**Comment by VitorGuizilini-TRI** [comment]
1) Yes, usually we need some fine-tuning on new datasets, so the pre-trained features can align with the new camera configurations. Creating camera-agnostic depth networks is a very interesting topic, though!
2) Depth accuracy degrades with distance, so far away objects like those found in the scenes you mentioned will suffer more. Repeating patterns are also an issue, the photometric loss will be ambiguous, so semi-supervision will help more in these cases. 
3) We assume that the input camera matrix corresponds to the original image resolution, and it's automatically scaled during resizing.

**Comment by Kartik17** [comment]
Thanks for your quick reply.

**Comment by tjdahlke** [comment]
Great questions @Kartik17.
